src/io/flutter/perf/FlutterWidgetPerf.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.perf;
+
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.intellij.concurrency.JobScheduler;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.TextEditor;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.ui.EdtInvocationManager;
+import groovy.util.MapEntry;
+import io.flutter.utils.AsyncUtils;
+
+import javax.swing.Timer;
+import java.awt.event.ActionEvent;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static io.flutter.inspector.InspectorService.toSourceLocationUri;
+
+public class FlutterWidgetPerf implements Disposable, Repaintable {
+
+  // Retry requests if we do not receive a response within this interval.
+  private static final long REQUEST_TIMEOUT_INTERVAL = 2000;
+
+  // Intentionally use a low FPS as the animations in EditorPerfDecorations
```

Comment:
Are we animating all windows or just visible ones?

Reply:
just ones that are visible.